### PR TITLE
Addition of sex loci

### PR DESCRIPTION
**Pull request by rnmitchell**
This PR will add annotation (custom, if necessary) for the loci located on the X & Y chromosomes assayed in both ForenSeq and PowerSeq. 

This includes:
* 7 loci on the X chromosome present only in the ForenSeq assay
* 17 loci on the Y chromosome present in both the ForenSeq and PowerSeq assays
* 3 loci on the Y chromosome present only in the PowerSeq assay
* 6 loci on the Y chromosome present only in the ForenSeq assay

### CODE REVIEW

**Comment by rnmitchell** [comment]
This is finally ready for review @standage! If you think there are additional tests I should add, let me know, happy to do that (although there's over 100 now! 😮 ).

**Comment by rnmitchell** [comment]
This ended up being a huge branch... sorry about that. I should have split it up more and will try to do so in the future!

File: lusSTR/marker.py [hunk 1]
```diff
@@ -99,13 +103,18 @@ def flankseq_5p(self):
...
     @property
     def flank_5p(self):
-        if self.uas:
+        if self.uas or self.flankseq_5p == '':
             return None
-        elif self.kit == 'powerseq' and self.data['Power_5'] > self.data['Foren_5']:
+        elif (
+            self.kit == 'powerseq' and self.data['Power_5'] > self.data['Foren_5']
+            and self.data['Foren_5'] > 0
+        ):
```
Comment (standage) [inline lusSTR/marker.py:117]: It's hard to look at this and nod my head and say "Ok, yeah, I understand what's going on here." Like, why is it *this particular set of conditions* that makes collapsing flanks in this way suitable for these loci?

The code does little to illuminate the reader's understanding of why particular loci are handled a certain way. To be clear, I don't think this is a reflection the coder. I understand this whole business of locus-specific conventions is very complex and fraught with substantial baggage and difficult to design for. Sometimes it seems like there are more exceptions than rules! If the code is providing the correct annotations, that may be the best we can expect at the moment.

File: lusSTR/marker.py [hunk 2]
```diff
@@ -1027,7 +1547,31 @@ def STRMarkerObject(locus, sequence, uas=False, kit='forenseq'):
...
+        'DXS8378': STRMarker_DXS8378,
+        'DXS7132': STRMarker_DXS7132,
+        'DXS10135': STRMarker_DXS10135,
+        'DXS10074': STRMarker_DXS10074,
+        'Y-GATA-H4': STRMarker_Y_GATA_H4,
+        'DYS390': STRMarker_DYS390,
+        'DYS385A-B': STRMarker_DYS385,
+        'DYS448': STRMarker_DYS448,
+        'DXS10103': STRMarker_DXS10103,
+        'DYS389II': STRMarker_DYS389II
```
Comment (standage) [inline lusSTR/marker.py:1574]: Just a few exceptions here...

😂😂😂   😭😭😭

File: lusSTR/tests/test_marker.py [hunk 3]
```diff
@@ -265,8 +265,475 @@ def test_strobj_D21S11_newformat():
...
+        'AATTTGGAAATTCTCCTCAGCATAACATTTTAATGATGATTCCTAGGATACAAGTGATGTGCTGAAAGTATCAATGTGTATCAGAAAA'
+        'CCAACATCTCTGCTTAGGTCTC', 'CAT [TCTA]4 TCCA [TCTA]6 CCTA CCTA CCTA [TCTA]2 TAGA [TCTA]3 '
+        'TCT', '11', '6', None, None, 'powerseq'
+    )
+])
+def test_strobj_Y_GATA_H4(sequence, bracketed, conc, lus, sec, tert, kit):
+    marker = STRMarkerObject('Y-GATA-H4', sequence, uas=False, kit=kit)
+    assert marker.annotation == bracketed
+    assert str(marker.canonical) == conc
+    assert marker.designation == (lus, sec, tert)
```
Comment (standage) [inline lusSTR/tests/test_marker.py:643]: In contrast, these tests do *a lot* to illuminate the reader's understanding. Kudos! I'm sure putting together these tests was tedious, and they'll be super valuable if any code refactoring is needed in the future.

**Review by standage — APPROVED** [review]
Forgive my brief lament about the complexity of the problems you're addressing here! This is looking pretty good, and the very robust test suite compensates for much of my concern about dense code in some places. Great job!

**Pull request merged**